test/base/routes/api-test.js

```diff
@@ -210,8 +213,10 @@ describe('routes: api', function() {
                 expect(options.headers.Authorization).to.equal('token ' + kgAuthToken);
             }
             spy.restore();
+            fakeClock.restore();
             done();
-        }, 0);
+        }, kgKernelRetentionTime);
+        fakeClock.tick(kgKernelRetentionTime + 1000);
     });
 });
 
```

Comment:
@nitind and I discussed this. Are any of the integration tests creating a real websocket client and testing communication through to a kernel gateway? (I didn't see one.) If not, then I don't think we should do all that at the moment just to get this in. As it stands, this is even hard to test manually when the browser and dashboard server are colocated on the same host. They really need to be on separate hosts with a network failure between them to see jupyter-js-services execute its reconnect logic.